receiver/k8seventsreceiver/receiver_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8seventsreceiver
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func TestNewReceiver(t *testing.T) {
+	rCfg := createDefaultConfig().(*Config)
+	client := fake.NewSimpleClientset()
+	r, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		consumertest.NewNop(),
+		client,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, r)
+	require.NoError(t, r.Start(context.Background(), componenttest.NewNopHost()))
+	assert.NoError(t, r.Shutdown(context.Background()))
+
+	rCfg.Namespaces = []string{"test", "another_test"}
+	r1, err := newReceiver(
+		componenttest.NewNopReceiverCreateSettings(),
+		rCfg,
+		consumertest.NewNop(),
+		client,
+	)
+
+	require.NoError(t, err)
+	require.NotNil(t, r1)
+	require.NoError(t, r1.Start(context.Background(), componenttest.NewNopHost()))
+	assert.NoError(t, r1.Shutdown(context.Background()))
+}
+
+func TestHandleNewEvent(t *testing.T) {
+	rCfg := createDefaultConfig().(*Config)
+	client := fake.NewSimpleClientset()
```

Comment:
Can we have a test case when we send couple of event objects to this fake client?

Reply:
It doesn't seem like we can add such test cases, because we are collecting (using 3rd party libraries) K8s events by making a REST API call, which needs an actual valid client with required data whereas the fake client we are using doesn't have an actual [REST client](https://github.com/kubernetes/client-go/blob/v0.23.1/kubernetes/typed/core/v1/fake/fake_core_client.go#L97) to make API calls for getting events.

Reply:
Yes but you can provide [fake objects](https://pkg.go.dev/k8s.io/client-go/kubernetes/fake#NewSimpleClientset) that will be returned by the client, no?

Reply:
It's true we can provide fake objects. In the case of events, if we [list](https://github.com/kubernetes/client-go/blob/v0.23.1/kubernetes/typed/core/v1/fake/fake_event.go#L58) or [watch](https://github.com/kubernetes/client-go/blob/v0.23.1/kubernetes/typed/core/v1/fake/fake_event.go#L80) events using that fake client, we'll get those events but the [library](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/6885/files#diff-7ae39264bf3789760f6289c310971d847b5768911a655b13ea44809408640d6aR126) we use to fetch events uses REST client. And the fake client doesn't have any implementation regarding the [REST client](https://github.com/kubernetes/client-go/blob/v0.23.1/kubernetes/typed/core/v1/fake/fake_core_client.go#L97), so providing fake objects will not work in our case.